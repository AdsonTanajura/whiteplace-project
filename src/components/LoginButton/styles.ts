import styled from 'styled-components';
import { SignInButton } from '@clerk/nextjs';

export const SignInButtonStyled = styled(SignInButton)`
  display: flex;
  background-color: #ffe492;
  border: none;
  align-items: center;
  text-align: center;
  font-size: 12px;
  padding: 8px 20px;
  border-radius: 8px;

  @media (min-width: 1024px) {
    min-height: 60px;
    max-width: 227px;
  }
`;
