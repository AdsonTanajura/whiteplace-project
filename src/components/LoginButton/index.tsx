import React from 'react';
import { SignedIn, SignedOut, UserButton } from '@clerk/nextjs';
import { SignInButtonStyled } from './styles';

const LoginButton = () => {
  return (
    <>
      <SignedOut>
        <SignInButtonStyled>Login</SignInButtonStyled>
      </SignedOut>
      <SignedIn>
        <UserButton />
      </SignedIn>
    </>
  );
};

export default LoginButton;
