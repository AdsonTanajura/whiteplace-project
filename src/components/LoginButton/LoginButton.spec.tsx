import { render } from '@testing-library/react';
import LoginButton from '../LoginButton';

describe('<LoginButton />', () => {
  it('should render login button', () => {
    const { queryAllByRole, container } = render(<LoginButton />);
    const buttons = queryAllByRole('button', { name: /Login/i });
    expect(buttons).toHaveLength(1);
    expect(container).toMatchSnapshot();
  });
});
