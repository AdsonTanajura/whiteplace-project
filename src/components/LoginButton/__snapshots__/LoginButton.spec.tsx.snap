// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginButton /> should render login button 1`] = `
<div>
  <button
    data-testid="clerk-sign-in-button"
  >
    Login
  </button>
  <button />
</div>
`;
